### PR TITLE
[TASK] Revise Database/ConnectionPool chapter

### PR DESCRIPTION
**Pull request by brotkrueml**
Mainly, get rid of the GeneralUtility::makeInstance() usage.
Also streamline wordings.

Releases: main, 11.5

### CODE REVIEW

File: Documentation/ApiOverview/Database/ConnectionPool/Index.rst [hunk 1]
```diff
@@ -1,56 +1,57 @@
...
-
-
-The `QueryBuilder` is the default workhorse object used by extension authors to express complex queries,
-while a `Connection` instance can be used as shortcut to deal with some simple query cases and little written down code.
+..  literalinclude:: _MyTableRepository.php
+    :caption: EXT:my_extension/Classes/Domain/Model/MyTableRepository.php
 
+The :php:`QueryBuilder` is the default workhorse object used by extension
+authors to express complex queries, while a :php:`Connection` instance can be
+used as a shortcut to handle some simple query cases and little written code.
```
Comment (linawolf) [inline Documentation/ApiOverview/Database/ConnectionPool/Index.rst:18]: This sentence sounds quite strange

File: Documentation/ApiOverview/Database/ConnectionPool/Index.rst [hunk 2]
```diff
@@ -1,56 +1,57 @@
...
+<database-configuration>` for details). This makes it possible to run tables on
+different databases without an extension developer having to worry about it.
+
+The :php:`ConnectionPool` implements this feature: It looks for configured
+table-to-database mapping and can return a :php:`Connection` or a
+:php:`QueryBuilder` instance for that specific connection. These objects know
+internally which target connection they are dealing with and will quote field
+names accordingly, for instance.
+
+Caveat
```
Comment (linawolf) [inline Documentation/ApiOverview/Database/ConnectionPool/Index.rst:35]: Where does this come from? Did you add the headline? What does it mean?

Comment (brotkrueml) [reply Documentation/ApiOverview/Database/ConnectionPool/Index.rst:35]: Yup, it's from me: "Caveat - a warning that particular things need to be considered before something can be done"
https://www.oxfordlearnersdictionaries.com/definition/english/caveat

Comment (linawolf) [reply Documentation/ApiOverview/Database/ConnectionPool/Index.rst:35]: Maybe the formulation is a little over the top as we are adressing ppl from all over the world, not just native speakers?

Comment (brotkrueml) [reply Documentation/ApiOverview/Database/ConnectionPool/Index.rst:35]: Okay, then "Beware" ;-)
I haven't seen this word quite some time. But I am open for suggestions ;-)

Comment (linawolf) [reply Documentation/ApiOverview/Database/ConnectionPool/Index.rst:35]: Beaware is good

File: Documentation/ApiOverview/Database/ConnectionPool/Index.rst (hunk 1 above)
Comment (brotkrueml) [reply Documentation/ApiOverview/Database/ConnectionPool/Index.rst]: Removed the last part ("and little written code").

**Review by linawolf — APPROVED** [review]

**Pull request merged**